website/docs/d/iam_policy_document.html.markdown

```diff
@@ -296,4 +202,94 @@ data "aws_iam_policy_document" "override_json_example" {
+  statement allows or denies the given actions. The default is "Allow".
+* `actions` (Optional) - A list of actions that this statement either allows
+  or denies. For example, ``["ec2:RunInstances", "s3:*"]``.
+* `not_actions` (Optional) - A list of actions that this statement does *not*
+  apply to. Used to apply a policy statement to all actions *except* those
+  listed.
+* `resources` (Optional) - A list of resource ARNs that this statement applies
+  to. This is required by AWS if used for an IAM policy.
+* `not_resources` (Optional) - A list of resource ARNs that this statement
+  does *not* apply to. Used to apply a policy statement to all resources
+  *except* those listed.
+* `principals` (Optional) - A nested configuration block (described below)
+  specifying a resource (or resource pattern) to which this statement applies.
+* `not_principals` (Optional) - Like `principals` except gives resources that
+  the statement does *not* apply to.
+* `condition` (Optional) - A nested configuration block (described below)
+  that defines a further, possibly-service-specific condition that constrains
+  whether this statement applies.
+
+Each policy may have either zero or more `principals` blocks or zero or more
+`not_principals` blocks, both of which each accept the following arguments:
+
+* `type` (Required) The type of principal. For AWS accounts this is "AWS".
+* `identifiers` (Required) List of identifiers for principals. When `type`
+  is "AWS", these are IAM user or role ARNs.
+
+Each policy statement may have zero or more `condition` blocks, which each
+accept the following arguments:
+
+* `test` (Required) The name of the
+  [IAM condition type](http://docs.aws.amazon.com/IAM/latest/UserGuide/reference_policies_elements.html#AccessPolicyLanguage_ConditionType)
+  to evaluate.
+* `variable` (Required) The name of a
+  [Context Variable](http://docs.aws.amazon.com/IAM/latest/UserGuide/reference_policies_elements.html#AvailableKeys)
+  to apply the condition to. Context variables may either be standard AWS
+  variables starting with `aws:`, or service-specific variables prefixed with
+  the service name.
+* `values` (Required) The values to evaluate the condition against. If multiple
+  values are provided, the condition matches if at least one of them applies.
+  (That is, the tests are combined with the "OR" boolean operation.)
+
+When multiple `condition` blocks are provided, they must *all* evaluate to true
+for the policy statement to apply. (In other words, the conditions are combined
+with the "AND" boolean operation.)
+
+## Context Variable Interpolation
+
+The IAM policy document format allows context variables to be interpolated
+into various strings within a statement. The native IAM policy document format
+uses `${...}`-style syntax that is in conflict with Terraform's interpolation
+syntax, so this data source instead uses `&{...}` syntax for interpolations that
```

Comment:
This suggests using `&{...}` syntax but L17 of the guide suggests using `$${...}`. Which one is officially recommended? It should probably be consistent.

Reply:
It looks like the `&{...}` format originated with the [datasource itself](https://github.com/hashicorp/terraform/pull/3124#issue-43710719) two years ago. As a TF user I think it would be good to settle on one of the two formats, though I agree mentioning the alternative format would be beneficial for users trying to understand an existing `aws_iam_policy_document` that may use the alternative format. I'm not sure if its worth deprecating one entirely, but it's something to consider.